tests/test_surface/test_deprecation.py

```diff
@@ -0,0 +1,61 @@
+import functools
+
+import pytest
+import deprecation
+from packaging import version
+
+import xtgeo
+
+
+def fail_if_not_removed(version_limit, msg=None):
+    if msg is None:
+        msg = "This method has reached its deprecation limit and must be removed"
+
+    def decorator(func):
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            func(*args, **kwargs)
+            if version.parse(xtgeo.version) > version.parse(version_limit):
+                pytest.fail(msg)
+
+        return wrapper
+
+    return decorator
```

Comment:
This functionality is already part of the `deprecation` library, and uses the actual version set in the decorator to deduce failure:
https://deprecation.readthedocs.io/en/latest/#using-fail-if-not-removed

Reply:
Yes, the reason is because the `deprecation` library uses a decorator to decorate functions, and `__init__` is not really deprecated, and when not using `deprecation`, we cant use the `fail_if_not_removed` , because I think it inspects that the message comes from the right function. So I made a custom wrapper for cases where we dont use `deprecation`

Reply:
And the message would not very very helpful: 
```
DeprecatedWarning: __init__ is deprecated as of 1.0. I am deprecated
```